x-pack/plugins/canvas/public/state/reducers/elements.js

```diff
@@ -10,6 +10,7 @@ import { get } from 'lodash';
 import * as actions from '../actions/elements';
 
 const getLocation = type => (type === 'group' ? 'groups' : 'elements');
+const firstOccurrence = (element, index, array) => array.indexOf(element) === index;
```

Comment:
This is a neat little de-duping helper, so rare to see that third arg used. 👍

nit: Since this is only used in 1 place, keeping it near where it's used is probably better for comprehension.

Reply:
Thanks @w33ble! IIRC there are a few other places in the layout engine that use such a function except they're inlined; if that's the case, I'd move it to `functional.js` and link to it from wherever else it's repeated now, otherwise move it near where it's used. Good you mentioned this rule of thumb, because otherwise I tended to put small, generic functions atop of files even if they were used only once.